dipa-service/src/main/java/de/muenchen/oss/swim/dipa/application/usecase/ProcessFileUseCase.java

```diff
@@ -0,0 +1,119 @@
+        try (InputStream fileStream = fileSystemOutPort.getPresignedUrlFile(event.presignedUrl())) {
+            switch (useCase.getType()) {
+            case HR_SUBFILE_INCOMING -> this.processHrSubfileIncoming(useCase, file, fileStream);
+            case null, default -> throw new IllegalStateException("UseCaseType not implemented");
+            }
+        } catch (final IOException e) {
+            throw new PresignedUrlException("Error while handling file InputStream", e);
+        }
+        // mark file as finished
+        fileEventOutPort.fileFinished(event);
+        log.info("File {} in use case {} finished", file, useCase.getName());
+        // update metric
+        dipaMeter.incrementProcessed(useCase.getName(), useCase.getType().name());
+    }
+
+    /**
+     * Process creation of Incoming with ContentObject inside HrSubfile.
+     * See {@link UseCaseType#HR_SUBFILE_INCOMING}.
+     *
+     * @param useCase UseCase to create the Incoming for.
+     * @param file File to create the Incoming for.
+     * @param fileContent Content of the ContentObject.
+     */
+    protected void processHrSubfileIncoming(final UseCase useCase, final File file, final InputStream fileContent) {
+        final HrSubfileContext requestContext = this.buildHrSubfileContext(useCase, file);
+        final IncomingRequest request = this.buildIncomingRequest(useCase, file, fileContent);
+        this.dipaOutPort.createHrSubfileIncoming(requestContext, request);
+    }
+
+    /**
+     * Build request object for creating an Incoming with a ContentObject.
+     *
+     * @param useCase UseCase to create the Incoming for.
+     * @param file File to create the Incoming for.
+     * @param fileContent Content of the ContentObject to create.
+     * @return Built Incoming request.
+     */
+    protected IncomingRequest buildIncomingRequest(final UseCase useCase, final File file, final InputStream fileContent) {
+        final String incomingSubject = this.patternHelper.applyPattern(useCase.getIncoming().getIncomingSubjPattern(), file.getFileNameWithoutExtension(),
+                null);
+        final ContentObjectRequest contentObject = this.buildContentObjectRequest(useCase, file, fileContent);
+        return new IncomingRequest(incomingSubject, contentObject);
+    }
+
+    /**
+     * Build request object for creating a ContentObject.
+     *
+     * @param useCase UseCase to create the ContentObject for.
+     * @param file File to create the ContentObject for.
+     * @param fileContent Content of the new ContentObject.
+     * @return Built ContentObject request.
+     */
+    protected ContentObjectRequest buildContentObjectRequest(final UseCase useCase, final File file, final InputStream fileContent) {
+        final String contentObjectName = String.format("%s.%s",
+                this.patternHelper.applyPattern(useCase.getContentObject().getFilenameOverwritePattern(), file.getFileNameWithoutExtension(), null),
```

Comment:
Not sure why, but doing this seems to create the contentObject name including the file extension (".pdf"):

![image](https://github.com/user-attachments/assets/44a61ff0-00b9-4c0b-b534-e157a6f67b50)

We must avoid that, because downloading the file from DIPA would result in a filename with extension ".pdf.pdf". 

Can you please check why this is happening?

Reply:
Done. I didn't think about that and just copied it from the dms-service but under SOAP the extension is provided separately.